### PR TITLE
Adds the ability to write report to the file

### LINKED ISSUE #9: report print to file
It would be great if there was a way to print the results directly to a file; csv would be ideal for starters.  This would make it easier to compare results over time to show what improvements have been made.

I could give a go at implementing this if you are open to the idea?

### PR DESCRIPTION
**Pull request by davydovanton**
Closed #9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,23 @@ allocated memory by file
 . . .
 ```
 
+Also yuo can print report to file with `print_to_file` method. By default this method create `memory_profiler.txt` file (or overwrite this file):
```
Comment (SamSaffron) [inline README.md]: typo yuo

File: README.md [hunk 2]
```diff
@@ -54,6 +54,23 @@ allocated memory by file
...
+```
+$ pry
+pry> require 'memory_profiler'
+pry> MemoryProfiler.report(allow_files: 'rubygems'){ require 'mime-types' }.print_to_file 'my_reposrt.txt'
```
Comment (SamSaffron) [inline README.md]: typo my_reposrt

File: lib/memory_profiler/results.rb [hunk 3]
```diff
@@ -97,6 +97,12 @@ def pretty_print(io = STDOUT, options = {})
       nil
     end
 
+    def print_to_file(file = 'memory_profiler.txt', options = {})
```
Comment (SamSaffron) [inline lib/memory_profiler/results.rb]: if we are to include this ... should be named pretty_print_to_file ... I am not 100% convinced we need it though. 

Maybe just change `pretty_print` to take in `io_or_filename`

**Comment by davydovanton** [comment]
@SamSaffron I fixed this :v:

File: README.md [hunk 4]
```diff
@@ -54,6 +54,23 @@ allocated memory by file
 . . .
 ```
 
+Also you can print report to file. For this use `pretty_print` method with `to_file` option and `path_to_your_log_file` string:
+```
+$ pry
+pry> require 'memory_profiler'
+pry> MemoryProfiler.report(allow_files: 'rubygems'){ require 'mime-types' }.pretty_print(to_file: 'path_to_your_log_file')
```
Comment (SamSaffron) [inline README.md]: param 1 is IO so this example will not work maybe just allow it to take string_or_io in ... its less to type anyway.

**Comment by davydovanton** [comment]
@SamSaffron I used double-splat syntax for `options` argument, what do you think about this solution?

**Comment by carloslopes** [comment]
:heart:

**Comment by SamSaffron** [comment]
cool thanks for this, double splat is an interesting solution here

**Comment by davydovanton** [comment]
@SamSaffron thanks for review :+1:

**Pull request merged**